src/Microsoft.DotNet.Interactive/InputTextStream.cs

```diff
@@ -46,7 +51,7 @@ private void EnsureStarted()
 
             Task.Run(async () =>
             {
-                while (!_complete)
+                while (!_complete && !_cancellationSource.IsCancellationRequested)
```

Comment:
Please don't do this to my CPU.

Reply:
it was fine on my Quantum processor but can see the point